spec/unit/coinbase/address/external_address_spec.rb

```diff
@@ -0,0 +1,202 @@
+        'Coinbase::Client::StakingContext::Context',
+        stakeable_balance: stake_balance,
+        unstakeable_balance: unstake_balance,
+        claimable_balance: claim_stake_balance
+      )
+    )
+  end
+
+  before(:each) do
+    allow(stake_api).to receive(:get_staking_context).and_return(staking_context)
+    allow(Coinbase::Client::StakeApi).to receive(:new).and_return(stake_api)
+  end
+
+  shared_examples 'it builds a staking operation' do |operation|
+    let(:transaction) { instance_double('Coinbase::Client::Transaction') }
+    let(:staking_operation) do
+      instance_double(
+        'Coinbase::Client::StakingOperation',
+        transaction: transaction
+      )
+    end
+
+    before(:each) do
+      allow(stake_api).to receive(:build_staking_operation).and_return(staking_operation)
+      allow(Coinbase::StakingOperation).to receive(:new)
+    end
+
+    it 'creates a new staking_api_client' do
+      subject
+      expect(Coinbase::Client::StakeApi).to have_received(:new)
+    end
+
+    it 'calls build_staking_operation' do
+      subject
+      expect(stake_api).to have_received(:build_staking_operation).with(
+        asset_id: asset_id,
+        address_id: id,
+        action: operation,
+        network_id: 'ethereum-mainnet',
+        options: { amount: (amount * Coinbase::WEI_PER_ETHER).to_s, mode: mode }
+      )
+    end
+
+    it 'creates a new StakingOperation' do
+      subject
+      expect(Coinbase::StakingOperation).to have_received(:new).with(staking_operation)
+    end
+
+    context 'when the mode is default' do
+      let(:mode) { :default }
+      context 'when the asset is :eth' do
+        let(:asset_id) { :eth }
+
+        it 'sets the mode to :partial' do
+          subject
+          expect(stake_api).to have_received(:build_staking_operation).with(
+            hash_including(options: hash_including(mode: :partial))
+          )
+        end
+      end
+    end
+
+    context "when the #{operation} balance is less than the amount" do
+      let(:"#{operation}_balance") { (amount - 1).to_s }
+
+      it 'raises an error' do
+        expect { subject }.to raise_error(Coinbase::InsufficientFundsError)
+      end
+    end
+
+    context 'when the amount is 0' do
+      let(:amount) { 0 }
+
+      it 'raises an error' do
+        expect { subject }.to raise_error(Coinbase::GreaterThanZeroError)
+      end
+    end
+
+    context 'when the amount is <0' do
+      let(:amount) { -1 }
+
+      it 'raises an error' do
+        expect { subject }.to raise_error(Coinbase::GreaterThanZeroError)
+      end
+    end
+  end
+
+  describe '#build_stake_operation' do
+    subject { address.build_stake_operation(amount, asset_id, mode: mode) }
+
+    it_behaves_like 'it builds a staking operation', 'stake'
+  end
+
+  describe '#build_unstake_operation' do
+    subject { address.build_unstake_operation(amount, asset_id, mode: mode) }
+    it_behaves_like 'it builds a staking operation', 'unstake'
```

Comment:
```suggestion

    it_behaves_like 'it builds a staking operation', 'unstake'
```

**non-blocking nit**: Will followup with rubocop-rspec, but would be nice to have consistent spacing here.

Feel free to let this get picked up by our rubocop-rspec subsequent pass!